File: src/Page/MainMenuPageAdvisor.java
package Page;

import SystemMessage.FunctionType;
import SystemMessage.SystemMessage;

public class MainMenuPageAdvisor extends Page {
	
	public MainMenuPageAdvisor(String content) {
		super(content);
		setType(PageType.MAIN_MENU_PAGE);
		setName("Main Menu Page Advisor");
	}
	
	@Override
	public SystemMessage runPage() {
		showContent();
		while (true) {
			switch (takeInput()) {
			case "1":
				return new SystemMessage(FunctionType.CHANGE_PAGE, PageType.PROFILE_PAGE, null);
			case "2":
				return new SystemMessage(FunctionType.CHANGE_PAGE, PageType.MY_STUDENTS_PAGE, null);
			case "3":
				return new SystemMessage(FunctionType.CHANGE_PAGE, PageType.EVALUATE_REQUESTS_PAGE, null);
			case "4":
				return new SystemMessage(FunctionType.CHANGE_PAGE, PageType.MY_COURSES_PAGE, null);
			case "5":
				return new SystemMessage(FunctionType.LOGOUT, PageType.LOGIN_PAGE, null);
			case "6":
				return new SystemMessage(FunctionType.EXIT, null, null);
			default:
				System.out.println("Wrong Input!");
			}
		}
	}
}

File: python_development/Page/MainMenuPageAdvisor.py
from python_development.Page.Page import Page
from python_development.Page.PageType import PageType
from python_development.System.FunctionType import FunctionType
from python_development.System.SystemMessage import SystemMessage

class MainMenuPageAdvisor(Page):
    
    def __init__(self, content):
        super().__init__(content)
        self.setType("MAIN_MENU_PAGE")
        self.setName("Main Menu Page Advisor")


    def runPage(self):
        self.showContent()
        while(True):
            inputFromPage = self.takeInput()

            if (inputFromPage == "1"):
                return SystemMessage("CHANGE_PAGE", "PROFILE_PAGE", None)
            elif (inputFromPage == "2"):
                return SystemMessage("CHANGE_PAGE", "MY_STUDENTS_PAGE", None)
            elif (inputFromPage == "3"):
                return SystemMessage("CHANGE_PAGE", "EVALUATE_REQUESTS_PAGE", None)
            elif (inputFromPage == "4"):
                return SystemMessage("CHANGE_PAGE", "MY_COURSES_PAGE", None)
            elif (inputFromPage == "5"):
                return SystemMessage("CHANGE_PAGE", "SYLLABUS_PAGE", None)
            elif (inputFromPage == "6"):
                return SystemMessage("LOGOUT", "LOGIN_PAGE", None)
            elif (inputFromPage == "7"):
                return SystemMessage("EXIT", None, None)
            else:
               print("Wrong Input!")
              